### PR TITLE
RUST-1440 bump bson clippy version to 1.63.0

### PR DESCRIPTION
**Pull request by abr-egn**

### CODE REVIEW

**Comment by isabelatkinson** [comment]
looks like there are still some clippy errors in CI. I think you've gotta run `cargo update` to see them locally, they're due to some new deprecations in `chrono`

**Comment by abr-egn** [comment]
Fixed the `chrono` deprecations - as a heads up, I was getting some confusing errors locally before I ran `cargo update` for `serde-tests` as well.

**Review by isabelatkinson — APPROVED** [review]

**Review by kmahar — APPROVED** [review]

File: src/datetime.rs [hunk 1]
```diff
@@ -168,7 +169,7 @@ impl crate::DateTime {
     ///
     /// let big = bson::DateTime::from_millis(i64::MAX);
     /// let chrono_big = big.to_chrono();
-    /// assert_eq!(chrono_big, chrono::MAX_DATETIME)
+    /// assert_eq!(chrono_big, chrono::DateTime::<chrono::Utc>::MAX_UTC)
```
Comment (patrickfreed) [inline src/datetime.rs:172]: Did chrono drop `MAX_DATETIME`? Or was this something clippy was complaining about?

File: src/raw/document_buf.rs [hunk 2]
```diff
@@ -386,7 +386,7 @@ impl Deref for RawDocumentBuf {
 
 impl Borrow<RawDocument> for RawDocumentBuf {
     fn borrow(&self) -> &RawDocument {
-        &*self
+        &**self
```
Comment (patrickfreed) [inline src/raw/document_buf.rs:389]: Why is the extra deref needed here?

File: src/datetime.rs (hunk 1 above)
Comment (abr-egn) [reply src/datetime.rs:172]: Chrono deprecated `MAX_DATETIME`, yeah.

File: src/raw/document_buf.rs (hunk 2 above)
Comment (abr-egn) [reply src/raw/document_buf.rs]: The error clippy made for this:
```
error: deref on an immutable reference
   --> src/raw/document_buf.rs:389:9
    |
389 |         &*self
    |         ^^^^^^
    |
    = note: `-D clippy::borrow-deref-ref` implied by `-D warnings`
    = help: for further information visit https://rust-lang.github.io/rust-clippy/master/index.html#borrow_deref_ref
help: if you would like to reborrow, try removing `&*`
    |
389 |         self
    |
help: if you would like to deref, try using `&**`
    |
389 |         &**self
    |
```

I've updated it to call `.deref()` directly for clarity.

**Review by patrickfreed — APPROVED** [review]

**Pull request merged**